### PR TITLE
Check size

### PR DESCRIPTION
**Pull request by rasapala**

### CODE REVIEW

**Review by dkalinowski — APPROVED** [review]

**Review by michalkulakowski — APPROVED** [review]

File: src/test/localfilesystem_test.cpp [hunk 1]
```diff
@@ -166,3 +166,50 @@ TEST(FileSystem, CheckIfPathIsEscaped) {
...
+    ASSERT_TRUE(ovms::FileSystem::isLocalFilesystem("o_O$az://"));
+    ASSERT_TRUE(ovms::FileSystem::isLocalFilesystem("../"));
+    ASSERT_TRUE(ovms::FileSystem::isLocalFilesystem("/localfilesystem"));
+    ASSERT_TRUE(ovms::FileSystem::isLocalFilesystem("/long/local/filesystem"));
+}
+
+TEST(FileSystem, setRootDirectoryPath) {
+    std::string rootPath = "";
+    std::string givenPath = "/givenpath";
+    auto fs = std::make_shared<ovms::LocalFileSystem>();
```
Comment (dkalinowski) [inline src/test/localfilesystem_test.cpp:187]: We should not be required to create LocalFileSystem class instance just to call `joinPath`. The function does not use any internal field, therefore it should be static.

What's more, joinPath exists in 2 classes - in LocalFileSystem and in AzureStorageAdapter - both methods are identical (and both non static).

File: src/modelmanager.cpp [hunk 2]
```diff
@@ -1311,7 +1311,7 @@ const std::string ModelManager::getFullPath(const std::string& pathToCheck) cons
     if (!FileSystem::isLocalFilesystem(pathToCheck)) {
         // Cloud filesystem
         return pathToCheck;
-    } else if (pathToCheck.at(0) == '/') {
+    } else if (pathToCheck.size() > 0 && pathToCheck.at(0) == '/') {
```
Comment (dkalinowski) [inline src/modelmanager.cpp:1314]: Missing unit test

File: src/filesystem.hpp [hunk 3]
```diff
@@ -177,7 +177,7 @@ class FileSystem {
         if (!FileSystem::isLocalFilesystem(givenPath)) {
             // Cloud filesystem
             path = givenPath;
-        } else if (givenPath.at(0) == '/') {
+        } else if (givenPath.size() > 0 && givenPath.at(0) == '/') {
```
Comment (dkalinowski) [inline src/filesystem.hpp:180]: Missing test

Comment (rasapala) [reply src/filesystem.hpp:180]: Added

**Pull request merged**